### PR TITLE
Add relative resolution to NXresolution, clean up ISO standard in NXphotoemission

### LINKED ISSUE #164: Add `relative_resolution` to NXresolution
We can avoid referencing two ISO fields in energy_resolution(NXresolution) as we have it now:

Overall energy resolution of the MPES instrument
This concept is related to term [10.7 ff.](https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:term:10.7) of the ISO 18115-1:2023 standard. -> points to `relative resolution of a spectrometer`
This concept is related to term [10.24](https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:term:10.24) of the ISO 18115-1:2023 standard. -> points to `energy resolution`

### PR DESCRIPTION
**Pull request by lukaspie**
This PR depends on #74. I made this its own PR so that we can review it better.

Resolves #164 and #166.

### CODE REVIEW

**Comment by lukaspie** [comment]
I wonder if we should split `NXenergydispersion\pass_energy` to `pass_energy` for hemispherics and `drift_energy` for TOFs. Currently, in `NXenergydispersion\pass_energy` we link to the [pass energy definition](https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:term:12.63) in the ISO standard which is _not_ the same as the TOF drift energy.

File: contributed_definitions/nyaml/NXsensor_scan.yaml
Comment (rettigl) [inline contributed_definitions/nyaml/NXsensor_scan.yaml:1]: This change does not seem to belong there.

**Review by rettigl — APPROVED** [review]
Technically, LGTM.
One could consider also adding the relative_resolution_errors at NXphotoemission and NXelectronanalyzer, even though not necessary (not different unit).

**Comment by rettigl** [comment]
> I wonder if we should split `NXenergydispersion\pass_energy` to `pass_energy` for hemispherics and `drift_energy` for TOFs. Currently, in `NXenergydispersion\pass_energy` we link to the [pass energy definition](https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:term:12.63) in the ISO standard which is _not_ the same as the TOF drift energy.

If we want to be strict with the terms, we can do that. Is there something for drift energy in ISO?

**Comment by lukaspie** [comment]
> One could consider also adding the relative_resolution_errors at NXphotoemission and NXelectronanalyzer, even though not necessary (not different unit).

Exactly, that's why I didn't add them, as that would have the same fields in the base class as in app def.



> > I wonder if we should split `NXenergydispersion\pass_energy` to `pass_energy` for hemispherics and `drift_energy` for TOFs. Currently, in `NXenergydispersion\pass_energy` we link to the [pass energy definition](https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:term:12.63) in the ISO standard which is _not_ the same as the TOF drift energy.
> 
> If we want to be strict with the terms, we can do that. Is there something for drift energy in ISO?

I did not find anything for drift energy, at least not in that particular ISO standard. I guess this was mostly developed with XPS and hemispherical analyzers in mind. I would still  argue it makes sense to split since they are distinct concepts, right?

**Comment by lukaspie** [comment]
If we add drift_energy, then we need to make pass_energy in NXphotoemission recommended. Does that work for you?

**Comment by domna** [comment]
> If we add drift_energy, then we need to make pass_energy in NXphotoemission recommended. Does that work for you?

Or we use a choice element for it? https://manual.nexusformat.org/defs_intro.html#choice

**Comment by lukaspie** [comment]
> Sorry, it seems NIAC has only intended this for groups.
So I guess it's the only choice to make it recommended.

Ok, makes sense. So both `pass_energy` and `drift_energy` recommended or just `pass_energy`?

**Comment by domna** [comment]
> > Sorry, it seems NIAC has only intended this for groups.
> > So I guess it's the only choice to make it recommended.
> 
> Ok, makes sense. So both `pass_energy` and `drift_energy` recommended or just `pass_energy`?

I would say both recommended with a note in the docstring that either one of them should be supplied, i.e., that their recommended for the respective technique.

**Comment by lukaspie** [comment]
Sounds good, I changed it to 
```yaml
pass_energy(NX_FLOAT):
  exists: recommended
  doc: |
    Either `pass_energy` or `drift_energy` must be supplied. `pass_energy` should be used when working
    with hemispherical analysers.
  unit: NX_ENERGY
drift_energy(NX_FLOAT):
  exists: recommended
  doc: |
    Either `pass_energy` or `drift_energy` must be supplied. `drift_energy` should be used if a TOF is used in the
    energy dispersive part of the electron analyzer.
  unit: NX_ENERGY
```

**Comment by lukaspie** [comment]
I have reverted the removal of `NXmpes` in this PR since we will have a discussion about naming the appdefs in the hierarchy again.

**Pull request merged**